onnx/test/shape_inference_test.py

```diff
@@ -154,8 +154,8 @@ def test_matmul_allow_unknown(self):  # type: () -> None
         self._make_matmul_test_allow_unknown((4, 2), (2, "a"), (4, "a"))
         self._make_matmul_test_allow_unknown((4, None), (2, "a"), (4, "a"))
         self._make_matmul_test_allow_unknown((4, None), (None, "a"), (4, "a"))
-        self._make_matmul_test_allow_unknown((1, 4, 2), ("a", 2, 5), ("a", 4, 5))
```

Comment:
the output "a" seems arbitrary - the input dims have 1 and "a" - so ideally, we need to stop at incrementing rank and not setting any value (actual value or symbolic)